src/Perpetuum/Services/Looting/CompositeLootGenerator.cs

```diff
@@ -11,6 +12,20 @@ public CompositeLootGenerator(params ILootGenerator[] generators)
             _generators = generators;
         }
 
+        public IEnumerable<LootGeneratorItemInfo> GetInfos()
+        {
+            var infos = _generators.Select(x => x.GetInfos());
+            var allInfos = new List<LootGeneratorItemInfo>();
+            foreach (var infoset in infos)
```

Comment:
If I'm reading this right, `_generators.Select(...)` generates an `IEnumerable<IEnumerable<LootGeneratorItemInfo>>`
In which case you can shorten it to this:
`infos.SelectMany(infoset => infoset).ToList();`

Also, as with the relic code, be careful passing IEnumerable around. You're not really passing the data, just a query pointer for deferred execution, the garbage collector will get really confused trying to keep these classes alive due to external pointing of a locally scoped object.

Reply:
With the .toList finalizer on the return, while it is returned as a IEnumerable, the promise of having data will have been fulfilled - is that a fair characterization?

Change as requested incoming